### PR TITLE
Module7

### PR DESCRIPTION
**Pull request by Arpitha-debug**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+on: [push]
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    environment: TodoApp_Environment
+    steps:
+    - uses: actions/checkout@v2
+    - run: echo Hello World
```
Comment (BenRamchandani) [inline .github/workflows/ci.yml]: Can delete this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    - name: Build Unit Tests
+      run: docker build --target test --tag todo-app:test .
+     
+
+    - name: Run Unit Test
+      run: docker run --env-file .env.test todo-app:test  /app/todo_app/tests
+
+    - name:  Build End to End tests
+      run: docker build --target test --tag todo-app:test .
```
Comment (BenRamchandani) [inline .github/workflows/ci.yml]: No need to rebuild the image

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      run: docker run -e TRELLO_BOARD_ID=${{ secrets.TRELLO_BOARD_ID }} -e TRELLO_TOKEN=${{ secrets.TRELLO_TOKEN }} -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e ID_LIST_NEWLIST=${{ secrets.ID_LIST_NEWLIST }} -e ID_LIST_TODO=${{ secrets.ID_LIST_TODO }} -e ID_LIST_DONE=${{ secrets.ID_LIST_DONE }} -e ID_LIST_DOING=${{ secrets.ID_LIST_DOING }} -e CARDS_TRELLODONE_CARD=${{ secrets.CARDS_TRELLODONE_CARD }} -e CARDS_TODO_CARD_TWO=${{ secrets.CARDS_TODO_CARD_TWO }} -e CARDS_TODO_CARD_THREE=${{ secrets.CARDS_TODO_CARD_THREE }} -e CARDS_TODO_CARD_ONE=${{ secrets.CARDS_TODO_CARD_ONE }} -e CARDS_DEVOPSTRELLOBOARD=${{ secrets.CARDS_DEVOPSTRELLOBOARD }} -e FLASK_APP=${{ secrets.FLASK_APP }} -e FLASK_ENV=${{ secrets.FLASK_ENV }} -e SECRET_KEY=${{ secrets.SECRET_KEY }}  -p 5100:5000 --mount type=bind,source="$(pwd)"/todo_app,target=/app/todo_app  todo-app:test /app/todo_app/SeleniumTest
+
+
+
+    -  name: Build and push
+       id: docker_build
+       uses: docker/build-push-action@v2
+       with:
+          push: true
+          tags: arpithadockerhub/dotnettemplate:latest
```
Comment (BenRamchandani) [inline .github/workflows/ci.yml]: That doesn't seem like the right tag!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+     
+
+    - name: Run Unit Test
+      run: docker run --env-file .env.test todo-app:test  /app/todo_app/tests
+
+    - name:  Build End to End tests
+      run: docker build --target test --tag todo-app:test .
+
+    - name: Run the end to end test
+      run: docker run -e TRELLO_BOARD_ID=${{ secrets.TRELLO_BOARD_ID }} -e TRELLO_TOKEN=${{ secrets.TRELLO_TOKEN }} -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e ID_LIST_NEWLIST=${{ secrets.ID_LIST_NEWLIST }} -e ID_LIST_TODO=${{ secrets.ID_LIST_TODO }} -e ID_LIST_DONE=${{ secrets.ID_LIST_DONE }} -e ID_LIST_DOING=${{ secrets.ID_LIST_DOING }} -e CARDS_TRELLODONE_CARD=${{ secrets.CARDS_TRELLODONE_CARD }} -e CARDS_TODO_CARD_TWO=${{ secrets.CARDS_TODO_CARD_TWO }} -e CARDS_TODO_CARD_THREE=${{ secrets.CARDS_TODO_CARD_THREE }} -e CARDS_TODO_CARD_ONE=${{ secrets.CARDS_TODO_CARD_ONE }} -e CARDS_DEVOPSTRELLOBOARD=${{ secrets.CARDS_DEVOPSTRELLOBOARD }} -e FLASK_APP=${{ secrets.FLASK_APP }} -e FLASK_ENV=${{ secrets.FLASK_ENV }} -e SECRET_KEY=${{ secrets.SECRET_KEY }}  -p 5100:5000 --mount type=bind,source="$(pwd)"/todo_app,target=/app/todo_app  todo-app:test /app/todo_app/SeleniumTest
```
Comment (BenRamchandani) [inline .github/workflows/ci.yml]: I don't think all of these are necessary for your tests to run. I needed

```
FLASK_APP
FLASK_ENV
SECRET_KEY
TRELLO_API_KEY
TRELLO_TOKEN
TRELLO_BOARD_ID
ID_LIST_TODO
ID_LIST_DOING
ID_LIST_DONE
```

**Review by BenRamchandani — APPROVED** [review]
Looks good! The most recent pipeline failed, but it looks like a transient error and I can see it has passed previously.

I've only reviewed the Module 7 CI pipeline here, let me know if there's anything else that needs reviewing now.

I've left some optional comments for tidying up the pipeline a bit, but everything works so go ahead and merge when you're ready.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: commented it out for now

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: Commented this as too

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: tag changed to app:latest

File: .github/workflows/ci.yml (hunk 4 above)
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: Removed others and run was successful after these changes

**Pull request merged**